service/src/main/kotlin/fi/espoo/evaka/shared/security/AccessControl.kt

```diff
@@ -732,6 +733,24 @@ WHERE employee_id = :userId
         )
     }
 
+    fun requirePermissionForEditingVasuFollowupEntry(user: AuthenticatedUser, id: VasuDocumentId, entryId: UUID) {
+        val globalRoles = when (user) {
+            is AuthenticatedUser.Employee -> user.globalRoles
+            else -> user.roles
+        }
+        if (globalRoles.any { it == UserRole.ADMIN })
+            return
+
+        Database(jdbi).connect { db ->
+            db.read { tx ->
+                val entry = tx.getVasuFollowupEntry(id, entryId)
+                if (entry.authorId != user.id && !user.hasOneOfRoles(*Action.VasuDocument.UPDATE_ANY_FOLLOWUP_ENTRY.defaultRoles().toTypedArray())) {
```

Comment:
We need to check both the roles and whether the user is the author (`user.id == authorId`) of a comment with id `entryId` in a vasu document with id `id`.

`ActionConfig` doesn't quite cut it when we need to compare this many IDs, and I don't think it's a good idea to make a generic solution for one instance of the problem.